spec/cucumber/formatter/junit_spec.rb

```diff
@@ -3,193 +3,249 @@
+            it { expect(@doc.to_s).not_to match /Things/ }
+            it { expect(@doc.to_s).not_to match /Good|Evil/ }
+            it { expect(@doc.to_s).not_to match /type="skipped"/}
+          end
 
-        define_feature <<-FEATURE
-          Feature: Eat things when hungry
-
-            Scenario Outline: Eat things
-              Given <Things>
-              And stuff:
-                | foo |
-                | bar |
-
-            Examples: Good
-              | Things   |
-              | Cucumber |
-              | Whisky   |
-            Examples: Evil
-              | Things   |
-              | Big Mac  |
-        FEATURE
-
-        it { expect(@doc.to_s).to match /Eat things when hungry/ }
-        it { expect(@doc.to_s).to match /Cucumber/ }
-        it { expect(@doc.to_s).to match /Whisky/ }
-        it { expect(@doc.to_s).to match /Big Mac/ }
-        it { expect(@doc.to_s).not_to match /Things/ }
-        it { expect(@doc.to_s).not_to match /Good|Evil/ }
-        it { expect(@doc.to_s).not_to match /type="skipped"/}
-      end
+          describe "scenario with skipped test in junit report" do
+            define_feature <<-FEATURE
+              Feature: junit report with skipped test
 
-      describe "scenario with skipped test in junit report" do
-        define_feature <<-FEATURE
-          Feature: junit report with skipped test
+                Scenario Outline: skip a test and junit report of the same
+                  Given a <skip> scenario
 
-            Scenario Outline: skip a test and junit report of the same
-              Given a <skip> scenario
+                Examples:
+                  | skip   |
+                  | undefined |
+                  | still undefined  |
+            FEATURE
 
-            Examples:
-              | skip   |
-              | undefined |
-              | still undefined  |
-        FEATURE
+            it { expect(@doc.to_s).to match /skipped="2"/}
+          end
 
-        it { expect(@doc.to_s).to match /skipped="2"/}
+          describe "with a regular data table scenario" do
+            define_steps do
+              Given(/the following items on a shortlist/) { |table| }
+              When(/I go.*/) {  }
+              Then(/I should have visited at least/) { |table| }
+            end
+
+            define_feature <<-FEATURE
+              Feature: Shortlist
+
+                Scenario: Procure items
+                  Given the following items on a shortlist:
+                    | item    |
+                    | milk    |
+                    | cookies |
+                  When I get some..
+                  Then I'll eat 'em
+
+            FEATURE
+            # these type of tables shouldn't crash (or generate test cases)
+            it { expect(@doc.to_s).not_to match /milk/ }
+            it { expect(@doc.to_s).not_to match /cookies/ }
+          end
+        end
       end
+      
+      context "In --expand mode" do
```

Comment:
This is where the new tests start.  Everything above was just moved into a context.